graphql/src/main/java/com/linecorp/armeria/server/graphql/GraphqlServiceBuilder.java

```diff
@@ -258,11 +280,18 @@ public GraphqlService build() {
             configurer.configure(builder);
         }
 
-        final DataLoaderRegistry dataLoaderRegistry = new DataLoaderRegistry();
-        final List<Consumer<? super DataLoaderRegistry>> dataLoaderRegistries =
-                dataLoaderRegistryConsumers.build();
-        for (Consumer<? super DataLoaderRegistry> configurer : dataLoaderRegistries) {
-            configurer.accept(dataLoaderRegistry);
+        final List<Consumer<? super DataLoaderRegistry>> dataLoaderRegistryConsumers =
+                this.dataLoaderRegistryConsumers.build();
+        if (dataLoaderRegistryFactory != null && !dataLoaderRegistryConsumers.isEmpty()) {
+            throw new IllegalStateException("Can't set dataLoaderRegistryFactory and " +
+                                            "dataLoaderRegistryConsumers together.");
+        }
```

Comment:
How about raising an exception as earlier as possible so that users can easily debug?
```java
@Nullable
private ImmutableList.Builder<Consumer<? super DataLoaderRegistry>> dataLoaderRegistryConsumers;

public GraphqlServiceBuilder dataLoaderRegistry(
        Function<? super ServiceRequestContext, ? extends DataLoaderRegistry> dataLoaderRegistryFactory) {
    checkArgument(dataLoaderRegistryConsumers == null,
                  "configureDataLoaderRegistry() and dataLoaderRegistry() are mutually exclusive.");
    this.dataLoaderRegistryFactory =
            requireNonNull(dataLoaderRegistryFactory, "dataLoaderRegistryFactory");
    return this;
}

public GraphqlServiceBuilder configureDataLoaderRegistry(
        Iterable<? extends Consumer<? super DataLoaderRegistry>> configurers) {
    checkArgument(dataLoaderRegistryFactory == null,
                  "configureDataLoaderRegistry() and dataLoaderRegistry() are mutually exclusive.");
    if (dataLoaderRegistryConsumers == null) {
        dataLoaderRegistryConsumers = ImmutableList.builder();
    }
    dataLoaderRegistryConsumers.addAll(requireNonNull(configurers, "configurers"));
    return this;
}
```

Reply:
Good idea 😎